Misc Changes to Schema

iskandarzulkarnaien opened the pull request:
Misc Changes to Schema

skyaxe97 committed:
updated schema

qjqqyy committed:
implement registration [2/2]

skyaxe97 committed:
created separate sql for trigger functions

qjqqyy committed:
fixup

qjqqyy committed:
we can login now

lock the (useless) about page behind auth wall so that we know logging
in works

skyaxe97 committed:
Created query sql, update create and trigger sql

skyaxe97 committed:
Update create

skyaxe97 committed:
Update triggers

qjqqyy commented on sql/create.sql:
```diff
@@ -1,37 +1,45 @@
-			mstatus = 'Member'
-			OR mstatus = 'Premium'
-			OR mstatus = 'VIP'
+			UPPER(mstatus) = 'MEMBER'
```
same here

qjqqyy commented on sql/create.sql:
```diff
@@ -1,37 +1,45 @@
-	CHECK (current_date - dob >= 16)				/* Check if user is 16 y/o and above */
+	CHECK (date_part('year', age(dob)) >= 16),		/* Check if user is 16 y/o and above */
+    CHECK (                                         /* Check gender */
+            UPPER(gender) = 'M'
```
if we're using the value like an enum `upper` in here confuses select statements

qjqqyy commented on sql/create.sql:
```diff
@@ -1,37 +1,45 @@
-	tpoints integer DEFAULT 100,					/* Accumulative reward points (upon joining: all members enjoy 100 points free) */
+	tpoints integer DEFAULT 100,					/* Total accumalated reward points (upon joining: all members enjoy 100 points free) */
+    cpoints integer DEFAULT 100,                    /* Current points user has after using points for Discount */
+	rating numeric (1,2) DEFAULT 5.00,				/* Default Passenger rating */
```
we can update this (and driver ratings) with triggers right? (aggregate on every update to Transactions)

skyaxe97 replied:
Added trigger

qjqqyy commented on sql/create.sql:
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
 	CHECK (											/* D = Discount, P = Point */
-			rcode LIKE 'D%'
-		   	OR rcode LIKE 'P%'
+			UPPER(rcode) LIKE 'D%'
```
same here

qjqqyy commented on sql/create.sql:
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
+	CHECK (											/* Passenger ratings: Range of 0 to 5 inclusive */
 			prating <= 5
-		   	OR prating >= 0
+		   	AND prating >= 0
```
good fix 👍

qjqqyy commented on sql/create.sql:
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
 
-CREATE TABLE Transacts (
+CREATE TABLE Transactions (
+	tcode varchar(15) PRIMARY KEY,
```
why not reference a Ride instead?
Doesn't a ride and passenger unambiguously identify a transaction?

qjqqyy commented on sql/create.sql:
```diff
@@ -41,155 +49,160 @@ CREATE TABLE Car (
+		  )
+);
+
+CREATE TABLE Payment (
```
this (in its current state) looks like an additional column in transactions

skyaxe97 committed:
updated create sql

skyaxe97 committed:
updated create,drop,triggers,dummy data

skyaxe97 committed:
Updated create,trigger and dummy input

skyaxe97 committed:
Updated create,trigger,dummy data

skyaxe97 committed:
Updated trigger,dummy data and drop

qjqqyy committed:
hash passwords in dummy data

qjqqyy committed:
Merge branch 'master' into sql-changes

qjqqyy committed:
remove uppercase in checks

otherwise the selection statement gets weird, consider
select blah from users where gender = 'M' or gender = 'm'
